### ISSUE TITLE
Launching `deluge-gtk` results in `ImportError` for `GLib`

### ISSUE THREAD

**Issue by Amar1729** [Amar1729/homebrew-deluge-meta#23]
Since various updates to formulae and dependencies (python3.12? libtorrent / pygobject bindings?), launching `deluge-gtk` now results in this error:

```
❯  deluge-gtk
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/deluge-meta/2.1.1_3/libexec/bin/deluge-gtk", line 8, in <module>
    sys.exit(start())
             ^^^^^^^
  File "/opt/homebrew/Cellar/deluge-meta/2.1.1_3/libexec/lib/python3.12/site-packages/deluge/ui/gtk3/__init__.py", line 64, in start
    Gtk().start()
  File "/opt/homebrew/Cellar/deluge-meta/2.1.1_3/libexec/lib/python3.12/site-packages/deluge/ui/gtk3/__init__.py", line 45, in start
    from .gtkui import GtkUI
  File "/opt/homebrew/Cellar/deluge-meta/2.1.1_3/libexec/lib/python3.12/site-packages/deluge/ui/gtk3/gtkui.py", line 22, in <module>
    from gi.repository.GLib import set_prgname
  File "/opt/homebrew/lib/python3.12/site-packages/gi/importer.py", line 133, in create_module
    raise ImportError('cannot import name %s, '
ImportError: cannot import name GLib, introspection typelib not found
```

However, running ` /opt/homebrew/opt/deluge-meta/libexec/bin/python -c 'import gi.repository'` successfully completes, so it's just `GLib` that's not being found? I need to track down what formula provides it: internet seems to indicate `gir1.2-gtk-3.0` on ubuntu, which of course isn't a thing on mac. this formula does depend on `gtk+3`, and yet something's still broken ...

**Comment by ForgottenFates7** [comment]
I had this same error after doing a `brew upgrade` yesterday at around 20:00 UTC, which upgraded my deluge-meta to the 2.1.1_3 version now at the HEAD of this repo.  Today I ran `brew doctor` and it told me:

```
Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  glib
```

When I ran `brew link`, the error went away but a new one came in its place:

```
$ deluge
07:54:48 [ERROR   ][deluge.ui.gtk3                    :52  ] 'gi.repository.Gio' object has no attribute 'content_type_get_icon'
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/deluge-meta/2.1.1_3/libexec/lib/python3.12/site-packages/deluge/ui/gtk3/__init__.py", line 49, in run
    gtkui = GtkUI(options)
            ^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/deluge-meta/2.1.1_3/libexec/lib/python3.12/site-packages/deluge/ui/gtk3/gtkui.py", line 200, in __init__
    self.torrentdetails = TorrentDetails()
                          ^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/deluge-meta/2.1.1_3/libexec/lib/python3.12/site-packages/deluge/ui/gtk3/torrentdetails.py", line 124, in __init__
    from .files_tab import FilesTab
  File "/opt/homebrew/Cellar/deluge-meta/2.1.1_3/libexec/lib/python3.12/site-packages/deluge/ui/gtk3/files_tab.py", line 47, in <module>
    G_ICON_DIRECTORY = Gio.content_type_get_icon('inode/directory')
                       ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.12/site-packages/gi/overrides/__init__.py", line 32, in __getattr__
    return getattr(self._introspection_module, name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.12/site-packages/gi/module.py", line 126, in __getattr__
    raise AttributeError("%r object has no attribute %r" % (
AttributeError: 'gi.repository.Gio' object has no attribute 'content_type_get_icon'
Traceback (most recent call last):
  File "/opt/homebrew/Cellar/deluge-meta/2.1.1_3/libexec/bin/deluge", line 8, in <module>
    sys.exit(start_ui())
             ^^^^^^^^^^
  File "/opt/homebrew/Cellar/deluge-meta/2.1.1_3/libexec/lib/python3.12/site-packages/deluge/ui/ui_entry.py", line 140, in start_ui
    ui.start()
  File "/opt/homebrew/Cellar/deluge-meta/2.1.1_3/libexec/lib/python3.12/site-packages/deluge/ui/gtk3/__init__.py", line 55, in start
    deluge.common.run_profiled(
  File "/opt/homebrew/Cellar/deluge-meta/2.1.1_3/libexec/lib/python3.12/site-packages/deluge/common.py", line 1355, in run_profiled
    return func(*args)
           ^^^^^^^^^^^
  File "/opt/homebrew/Cellar/deluge-meta/2.1.1_3/libexec/lib/python3.12/site-packages/deluge/ui/gtk3/__init__.py", line 49, in run
    gtkui = GtkUI(options)
            ^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/deluge-meta/2.1.1_3/libexec/lib/python3.12/site-packages/deluge/ui/gtk3/gtkui.py", line 200, in __init__
    self.torrentdetails = TorrentDetails()
                          ^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/deluge-meta/2.1.1_3/libexec/lib/python3.12/site-packages/deluge/ui/gtk3/torrentdetails.py", line 124, in __init__
    from .files_tab import FilesTab
  File "/opt/homebrew/Cellar/deluge-meta/2.1.1_3/libexec/lib/python3.12/site-packages/deluge/ui/gtk3/files_tab.py", line 47, in <module>
    G_ICON_DIRECTORY = Gio.content_type_get_icon('inode/directory')
                       ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.12/site-packages/gi/overrides/__init__.py", line 32, in __getattr__
    return getattr(self._introspection_module, name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.12/site-packages/gi/module.py", line 126, in __getattr__
    raise AttributeError("%r object has no attribute %r" % (
AttributeError: 'gi.repository.Gio' object has no attribute 'content_type_get_icon'


```

**Comment by Amar1729** [comment]
Oh dang, good call on `brew doctor`, it looks like `glib` was also unlinked for me (how did that happen??). Yeah, i now see the same error as you when running `deluge`/`deluge-gtk`, which i assume is a dependency version problem somewhere. might have to pin something.

**Comment by Amar1729** [comment]
Ok, I think this is actually because something has changed in upstream `gobject-introspection`, as filed in this brew issue: https://github.com/Homebrew/homebrew-core/issues/167524

So, waiting on that fix before this can be addressed. It's also somewhat non-trivial to test locally, because gobject-introspection won't currently build locally for me because i have the recent commandline tools update.

**Comment by Amar1729** [comment]
Ok, i've fleshed out the required process for anybody running into this issue. It's a bit involved, but boils down to the following:
1. remove any packages that depend on `glib`/`gobject-introspection`
    - this is necessary because `gobject-introspection` writes some files to `glib`'s keg location.
2. rewind to a past commit in `homebrew/homebrew-core`
3. install `deluge-meta`, which will use the last working `gobject-introspection` (`v1.78.1)
4. pin `gobject-introspection` and reinstall your packages
5. fast-forward back to current and operate as normal

# 1: remove any leaves dependent on `glib`

This step is helpful if you (like me) have several packages dependent on `glib`. We will later use the output to re-install anything you uninstall here.

```bash
brew uninstall deluge-meta
brew autoremove
comm -1 -2 <(brew leaves | sort) <(brew uses --installed glib | sort) > pkgs.txt
comm -1 -2 <(brew leaves | sort) <(brew uses --installed gobject-introspection | sort) >> pkgs.txt

# look at the output of pkgs.txt (should be one package per line), then uninstall all of them:
cat pkgs.txt | xargs brew uninstall
brew autoremove

# this should succeed, since we've now removed anything that depends on glib (including gobject-introspection).
brew uninstall glib
```

2. rewind to last commit for `gobject-introspection` `v1.78.1`

```bash
# ensure you have homebrew/homebrew-core tapped locally:
brew tap homebrew/homebrew-core
# then rewind it to here:
# (you can find this commit yourself by doing `git log -- Formula/g/gobject-introspection.rb` in its git directory, and looking for the last commit that uses gobject-introspection 1.78.1)
git -C "$(brew --prefix)"/Library/Taps/homebrew/homebrew-core checkout abdcdb4613d762c58b0d59ae97f34e92e00f493d
```

3. install `deluge-meta`

IMPORTANT: during this step, you must set an environment variable to ensure homebrew doesn't by-default update your dependencies to their current versions based on an API call (it does this by default because it generally is a latest-version-only package manager).

```bash
# delete the cached data of up-to-date packages (just for safety, i dont think this is actually necessary)
rm "$(brew --cache)"/api/formula.jws.json

# now install
# keep an eye on what gets installed as this runs:
# it SHOULD install gobject-introspection 1.78.1 and glib 2.80.0.
HOMEBREW_NO_INSTALL_FROM_API=1 brew install amar1729/deluge-meta/deluge-meta

# test that this works by launching the GUI:
deluge # or deluge-gtk
```

4. reinstall and pin

This will stop `gobject-introspection` from being upgraded by homebrew, but importantly it will **also stop all of its dependents from being upgraded**. You will likely want to unpin this as soon as possible (when the object issue is fixed).

```bash
cat pkgs.txt | HOMEBREW_NO_INSTALL_FROM_API=1 xargs brew install

brew pin gobject-introspection glib
# there may be other formulae to pin here, i'm not sure ... i'm worried you have to pin python@3.12 itself, which would be annoying
```

5. fast-forward and operate as normal

You may restore the homebrew/homebrew-core tap to normal with:

```bash
git -C "$(brew --prefix)"/Library/Taps/homebrew/homebrew-core checkout abdcdb4613d762c58b0d59ae97f34e92e00f493d
```

Or, you may delete it: `brew untap homebrew/homebrew-core`. Any future brew operations will go back to using the API (unless you set `HOMEBREW_NO_INSTALL_FROM_API` in your shell profile).

**Comment by Amar1729** [comment]
PS. I don't really understand why, but homebrew seems to lose track of the versions of pinned packages. After resetting back to current homebrew/homebrew-core `origin/master` and running `brew upgrade --formulae --dry-run`, `brew` seems to think that i have `glib 2.80.0_2, gobject-introspection 1.80.1` (the current versions) even though i pinned at `2.80.0` and `1.78.1`, respectively.

run `brew upgrade` at your peril! 😬

**Comment by Amar1729** [comment]
Came up with a bit of a hacky two-liner to find all packages safe to upgrade:

```bash
# takes more time if you have more things dependent on glib (~1min for me)
brew uses --installed --recursive glib | tee glib-related.txt | xargs -n1 brew deps | sort -u >> glib-related.txt
# inspect the final output and see what brew says it will upgrade
# if you're ok with it you can re-run and remove --dry-run, or do it manually with brew upgrade pkg1 [pkg2 ...]
comm -1 -3 <(sort glib-related.txt) <(brew outdated --formula | awk '{print $1}' | grep -v "^glib$" | sort) | xargs brew upgrade --dry-run
```

**Comment by cipri-tom** [comment]
I ran the two-liner, while `deluge-meta` was installed, but it did not upgrade `deluge-meta` 🤔

**Comment by Amar1729** [comment]
@cipri-tom that makes sense: `deluge-meta` should be marked as unsafe to upgrade since it depends on `glib`. Besides, there are no upgrades for it yet (if you've done the hacky installation process above, you should have the most up-to-date version).

**Comment by 2600box** [comment]
Hello, I have been frustrated by this problem. I have tried following all the steps outlined in the different posts of this thread without success. Whatever I do, it still goes for the latest package.

Is there any further guidance or advice that could help?

I know this will be fixed in brew eventually, but I checking everyday, so getting the workaround to _work_ would help me a lot.

Thanks!

**Comment by cipri-tom** [comment]
I don't know if the fix in brew will be anytime soon. The issue is closed. They [deferred to Gnome](https://gitlab.gnome.org/GNOME/glib/-/issues/3322), which doesn't seem to have a resolution or action plan. To it may be a while before a fix lands

**Comment by vonProteus** [comment]
as far i know brew now use its api to determine current version of package
have you set environment variable `HOMEBREW_NO_INSTALL_FROM_API=1` to use git?

**Comment by 2600box** [comment]
> as far i know brew now use its api to determine current version of package have you set environment variable `HOMEBREW_NO_INSTALL_FROM_API=1` to use git?

Yes, I copy and paste the commands, though some modifications were required.

For example, I need to add `--force` `brew tap homebrew/homebrew-core`
and then 
I need to add `/Homebrew` before Library in `git -C "$(brew --prefix)"/Library/Taps/homebrew/homebrew-core checkout abdcdb4613d762c58b0d59ae97f34e92e00f493d`

other than that, running `HOMEBREW_NO_INSTALL_FROM_API=1 brew install amar1729/deluge-meta/deluge-meta` installs, but it uses 1.80.1 of gobject-introspection...

Am I missing a step? Thanks for any guidance here


```
➜  ~ brew uninstall deluge-meta
Error: No such keg: /usr/local/Cellar/deluge-meta
➜  ~ brew autoremove
➜  ~ comm -1 -2 <(brew leaves | sort) <(brew uses --installed glib | sort) > pkgs.txt
➜  ~ comm -1 -2 <(brew leaves | sort) <(brew uses --installed gobject-introspection | sort) >> pkgs.txt
➜  ~ cat pkgs.txt | xargs brew uninstall
➜  ~ brew autoremove
➜  ~ brew uninstall glib
Error: No such keg: /usr/local/Cellar/glib
➜  ~ brew tap homebrew/homebrew-core --force
➜  ~ !git:p
➜  ~ git -C "$(brew --prefix)"/Homebrew/Library/Taps/homebrew/homebrew-core checkout abdcdb4613d762c58b0d59ae97f34e92e00f493d
Previous HEAD position was 5f1509692b4 Merge pull request #170182 from chromy/add-task2
HEAD is now at abdcdb4613d gobject-introspection: update 1.78.1 bottle.
➜  ~ rm "$(brew --cache)"/api/formula.jws.json
```

```
➜  ~ HOMEBREW_NO_INSTALL_FROM_API=1 brew install amar1729/deluge-meta/deluge-meta
==> Auto-updating Homebrew...
Adjust how often this is run with HOMEBREW_AUTO_UPDATE_SECS or disable with
HOMEBREW_NO_AUTO_UPDATE. Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
==> Auto-updated Homebrew!
Updated 1 tap (homebrew/core).
==> New Formulae
autotrace                     ffmpeg@6                      liblc3                        msieve                        rage                          tenv
beakerlib                     git-split-diffs               libscfg                       navidrome                     ratchet                       uni-algo
beancount-language-server     gitu                          libtickit                     oj                            redict                        uutils-diffutils
c-blosc2                      ingress2gateway               llvm@17                       overarch                      rtabmap                       valkey
cotp                          jnv                           logdy                         parsedmarc                    rustcat                       vfox
descope                       jtbl                          manim                         pnpm@8                        superfile                     whisperkit-cli
dissent                       kubecolor                     mantra                        policy_sentry                 sysaidmin                     yo
dpcmd                         kuto                          mdformat                      promptfoo                     tartufo
espeak-ng                     lexido                        morpheus                      protoc-gen-js                 task@2

You have 5 outdated formulae installed.

==> Fetching dependencies for amar1729/deluge-meta/deluge-meta: glib, libxrender, lzo, pixman, cairo, gdk-pixbuf, fribidi, graphite2, harfbuzz, pango, librsvg, adwaita-icon-theme, pycparser, cffi, cryptography, dbus, libxfixes, libxi, libxtst, at-spi2-core, gsettings-desktop-schemas, hicolor-icon-theme, libepoxy, gtk+3, gobject-introspection, py3cairo, pygobject3, python-setuptools, python-typing-extensions, six and zlib
==> Downloading https://ghcr.io/v2/homebrew/core/glib/manifests/2.80.0_2
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/bad07197fbf4d9c1af0f49bb79d89a3b369de9b90872243c8bbcfea7d4385475--glib-2.80.0_2.bottle_manifest.json
==> Fetching glib
==> Downloading https://ghcr.io/v2/homebrew/core/glib/blobs/sha256:c38a32e98f6cd669499a3deb1a7b220cb9de04300471537ac22f27f2e6aa3d35
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/9d1ca951c83a697ffb55091c2d6d6d4bbe42e2ba593542c9d9267f335133007d--glib--2.80.0_2.ventura.bottle.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/libxrender/manifests/0.9.11
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/134c11be8346a1b116e04983c2da6366f29c4f4c2abc17604dcdb80d0475ae9d--libxrender-0.9.11.bottle_manifest.json
==> Fetching libxrender
==> Downloading https://ghcr.io/v2/homebrew/core/libxrender/blobs/sha256:5eb0bb57e683a00176a5702bdf021641fe9290e84e0414d2979823ad0aad1d3a
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/b604b27ded2aa76c040ef0a636f996439ab5e28b90e3332d237f7f28659b4bdd--libxrender--0.9.11.ventura.bottle.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/lzo/manifests/2.10
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/d4aa5b0c239912c53bc857d1012c6b7feb4acb509618f5e100f95bf8521f08e7--lzo-2.10.bottle_manifest.json
==> Fetching lzo
==> Downloading https://ghcr.io/v2/homebrew/core/lzo/blobs/sha256:ac88f2fdcb7eb5f82e1e6b2459408f2ca3db299d2366b7af64e410a3c6629b52
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/ddae0c352b464663ac79ffa18344ff043ffb799ca6c8912d48c22b77b40a478d--lzo--2.10.ventura.bottle.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/pixman/manifests/0.42.2-1
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/2b0d8f2c63712ce503fa54309b732051a7ca5cd31bd561aa4aaba331bf26bcef--pixman-0.42.2-1.bottle_manifest.json
==> Fetching pixman
==> Downloading https://ghcr.io/v2/homebrew/core/pixman/blobs/sha256:84c3bfc0a0e43b714fd064954885314b4ec2928571ba43c49760cacca50bd32c
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/da5a5dbb8e94eb96f81b0367d1d1268944ff531442f1064733dd52f84b1d1c8b--pixman--0.42.2.ventura.bottle.1.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/cairo/manifests/1.18.0
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/1b8b7f19e38cb0e029497dfe0ad3f3557f605025c5575bdfe86c17b889c6e6d1--cairo-1.18.0.bottle_manifest.json
==> Fetching cairo
==> Downloading https://ghcr.io/v2/homebrew/core/cairo/blobs/sha256:53fa7ded83d0f45fd7c8c25fedb970f1084bc4861f10988f36c2f2cfd6064552
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/fbe56ed54050d0718898152e5fcbc097df71a41700a5a65dc14432dfcd6ba67d--cairo--1.18.0.ventura.bottle.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/gdk-pixbuf/manifests/2.42.11-1
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/e33b7eccf9c79f017289001438f9402b37bc235e3dc91458dd57345ad2ed1098--gdk-pixbuf-2.42.11-1.bottle_manifest.json
==> Fetching gdk-pixbuf
==> Downloading https://ghcr.io/v2/homebrew/core/gdk-pixbuf/blobs/sha256:9a78d1735066a77f229532ce0ccb5a463c4fb31fd5e9c54c4daa2cf3840795da
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/f2a1c8c17b308b75efa29d9a75222fd24d5a65ac008dd03e9c9de0d0961084f6--gdk-pixbuf--2.42.11.ventura.bottle.1.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/fribidi/manifests/1.0.14
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/e6b501ebd099a81598298a9fe97eac544af056e9b6031dabf29d953431e229c1--fribidi-1.0.14.bottle_manifest.json
==> Fetching fribidi
==> Downloading https://ghcr.io/v2/homebrew/core/fribidi/blobs/sha256:c8b57d7acc7618677023c3849095b7f89e6440b747ae589704d30a607e0c6d56
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/f1da4648ae9c292a2cfee23b271af24868ffde583dc82c783ac9fe7a9a7d44b8--fribidi--1.0.14.ventura.bottle.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/graphite2/manifests/1.3.14
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/bbb4dd2ef1846301d1eb08053e19e11ca9c780f93f4d3b2d638fd94a9bf54a0c--graphite2-1.3.14.bottle_manifest.json
==> Fetching graphite2
==> Downloading https://ghcr.io/v2/homebrew/core/graphite2/blobs/sha256:db73b7ef0318611d6dcf795cd4e43c5b62c5798190bc634dc94c6530e35afc46
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/c725d86366abee2ec29698d4bccebcfe81f7ebc42b3f45063a992af311c7903b--graphite2--1.3.14.ventura.bottle.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/harfbuzz/manifests/8.4.0
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/fd47e37d9f5b5084316ec51955470aa32e038246275985aee10b1881a4f7344b--harfbuzz-8.4.0.bottle_manifest.json
==> Fetching harfbuzz
==> Downloading https://ghcr.io/v2/homebrew/core/harfbuzz/blobs/sha256:c347ae63119c1d8484f7711a17d8973f50194c215970b3ecccbfba186f70cce8
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/d9bbd6e1938d7ec9e43967d6eba35e104bb36e9b67b91b815f2c1d0e8a471ff1--harfbuzz--8.4.0.ventura.bottle.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/pango/manifests/1.52.2
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/214b7cb46e8e1680ea60d3578939e548b2daf2721164a11efeda086b9d494949--pango-1.52.2.bottle_manifest.json
==> Fetching pango
==> Downloading https://ghcr.io/v2/homebrew/core/pango/blobs/sha256:a4f074c18620090f87aa15b1879126cf9cb2f5ddaaa2c0b54e2fd17cd591485d
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/f158b96af87bca2ab00d751b972fe6b357c66ac529f884b3b32baa3d54121fb4--pango--1.52.2.ventura.bottle.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/librsvg/manifests/2.58.0
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/7b1ad23688052396ca6401b20573b2f69fafc28221b623268521dc99ef161ea7--librsvg-2.58.0.bottle_manifest.json
==> Fetching librsvg
==> Downloading https://ghcr.io/v2/homebrew/core/librsvg/blobs/sha256:8b30ecc1378e8e4f791a8d3e8e6473963f6ae3346a1bc3aa199d019b30f291b2
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/8118162dda092c8860c0dcacb0a137079fa70b04d401e1344e0e6b93eea839a3--librsvg--2.58.0.ventura.bottle.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/adwaita-icon-theme/manifests/46.0
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/696462fef1577dada7d1fc747b9b32c68590e48b3d1328f7dbd07c8a90d31132--adwaita-icon-theme-46.0.bottle_manifest.json
==> Fetching adwaita-icon-theme
==> Downloading https://ghcr.io/v2/homebrew/core/adwaita-icon-theme/blobs/sha256:380e2eac20713968f6e73603bfaeab035d0c6d07a1fa16e431decbc3c8aeb1cd
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/d36e8ad81224a8a1cb3a38ac06ba498ee686fbae5d575b408bb58575476b694c--adwaita-icon-theme--46.0.all.bottle.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/pycparser/manifests/2.22
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/eeea4f104b2e432de9839123571bf8ac17b912aa138005099f4e97992d3e56b3--pycparser-2.22.bottle_manifest.json
==> Fetching pycparser
==> Downloading https://ghcr.io/v2/homebrew/core/pycparser/blobs/sha256:493634a14bdd3491626f9b76d04161cdaa045165fb8efffa9bf780efd823d844
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/c2b9a642a68275062876982dd838ecc35d160d3d5a9a3af56a58898985cd1d61--pycparser--2.22.ventura.bottle.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/cffi/manifests/1.16.0_1-1
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/eb1e9d6f4f724404e336e6bd8af635bd56b992b0d8a67c5665a2ed7f91938b46--cffi-1.16.0_1-1.bottle_manifest.json
==> Fetching cffi
==> Downloading https://ghcr.io/v2/homebrew/core/cffi/blobs/sha256:0bd893c328706682088abca867d52c40b8a9e162e65c5f067647cf3fe4171821
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/3db74983ee74a756664eeabe8d11517f921c3eb7117d52ee39061e126b486ce4--cffi--1.16.0_1.ventura.bottle.1.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/cryptography/manifests/42.0.5
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/e1105ccd58b6027d55214410024349f2af1f04f2cd983a527a6757e5fa86179f--cryptography-42.0.5.bottle_manifest.json
==> Fetching cryptography
==> Downloading https://ghcr.io/v2/homebrew/core/cryptography/blobs/sha256:8d208effe9f9c00d4742aa5f7c60f9793a69b6396c5df05e0a8c0c776035d060
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/2e66d8b3e9b1b1c0e21284295375b25d9ec0ab84fcf5ac699e0ed04fc31d85eb--cryptography--42.0.5.ventura.bottle.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/dbus/manifests/1.14.10
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/2dee97539c09d15349d9935f14c299bce3e52d11fd6d399a05a5190d0e7a1028--dbus-1.14.10.bottle_manifest.json
==> Fetching dbus
==> Downloading https://ghcr.io/v2/homebrew/core/dbus/blobs/sha256:6ed57658615731eac10b392a02031bd9e42025764bb806070c7acfea86bd8e5d
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/3ff7cf0938f64cbe2a4228a8bc84458d5ac5e9a8b2f620d9d911fe6747f86671--dbus--1.14.10.ventura.bottle.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/libxfixes/manifests/6.0.1
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/b78eeb464b4c153eb6e75a04ba240b4972cdc642e77e9c5085e186630c2c4862--libxfixes-6.0.1.bottle_manifest.json
==> Fetching libxfixes
==> Downloading https://ghcr.io/v2/homebrew/core/libxfixes/blobs/sha256:ade02ac4b73db0272d8bdb95bd05f8c8c11683daa944c66a10f1e72740bac364
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/9c05e05722956c1e8f6f4c7cedae9e43747ad9fbdccc5380232983bdb6ab152e--libxfixes--6.0.1.ventura.bottle.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/libxi/manifests/1.8.1
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/fd0ed36e5bf4048d89c5a4691f66ea6d88f113822452224af6ada7c579428ab0--libxi-1.8.1.bottle_manifest.json
==> Fetching libxi
==> Downloading https://ghcr.io/v2/homebrew/core/libxi/blobs/sha256:7b3506e3d213cc91362a9b97f8644d0c5084f963b09c1d9e76cb733374f0e042
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/53872e7804a1428ba45afc6996bde01cd1d605b37776fae2018691c7be7108f8--libxi--1.8.1.ventura.bottle.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/libxtst/manifests/1.2.4
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/c3624db58ac89924c9a01073cf5dc8a8d1acbe9c3ed52af7ec5a74e92816d50c--libxtst-1.2.4.bottle_manifest.json
==> Fetching libxtst
==> Downloading https://ghcr.io/v2/homebrew/core/libxtst/blobs/sha256:e9aa2923a5647444b5557ae761abdc05730c5c0bd836103853921ef9df38d4dc
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/c346142f7646be6378059838b261765ce4006e2a76f96c712b7d632647cada97--libxtst--1.2.4.ventura.bottle.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/at-spi2-core/manifests/2.52.0
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/32108c66e5f0d77340b05ddf14af03f14415a6c4490ade88b0e5d27edc0de1eb--at-spi2-core-2.52.0.bottle_manifest.json
==> Fetching at-spi2-core
==> Downloading https://ghcr.io/v2/homebrew/core/at-spi2-core/blobs/sha256:b3d3cc4309862f55efd46a713f1ba2dd813be0f654f70525291f2a631f1d35f7
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/365d79e3b850181231a0ef6f8a14836a9064128d2517a5b43f81479a261da58c--at-spi2-core--2.52.0.ventura.bottle.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/gsettings-desktop-schemas/manifests/46.0
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/9b0da63c1cb9e0d2c49b1e1ab0db3278275b969537c1dc3a6d8f663115f8310e--gsettings-desktop-schemas-46.0.bottle_manifest.json
==> Fetching gsettings-desktop-schemas
==> Downloading https://ghcr.io/v2/homebrew/core/gsettings-desktop-schemas/blobs/sha256:0cf5db7c142d6769c09befe16e21caed32f43595367752be53141fd8b0820f41
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/41b3f170dd13fef06a72a900608309e371b990242aeecf43888ee689a7d9427d--gsettings-desktop-schemas--46.0.ventura.bottle.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/hicolor-icon-theme/manifests/0.17
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/713c984987cfb15e35fc3fa0e951f5ab343f2154fdeab925615cb11b5bef9389--hicolor-icon-theme-0.17.bottle_manifest.json
==> Fetching hicolor-icon-theme
==> Downloading https://ghcr.io/v2/homebrew/core/hicolor-icon-theme/blobs/sha256:fca72dd306b23edae0d4d66b0bd485e66f53f175ba1ca87a6167309d73533bc8
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/91e26877be8952c6d3d8b8340b788f89af4d4ea0080c2e939b7a65a737c1f92e--hicolor-icon-theme--0.17.ventura.bottle.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/libepoxy/manifests/1.5.10
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/fd91321dc6032305294ad3c7c3851bc80f1505f18512e4753036c509ba4b4e2c--libepoxy-1.5.10.bottle_manifest.json
==> Fetching libepoxy
==> Downloading https://ghcr.io/v2/homebrew/core/libepoxy/blobs/sha256:91cca5d118a350e7105a303fd873915fc5f36c0a83be02101f3c742a52d0059c
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/65c3e6f8c304491477aea9c4e5f845b969ece8d965daee91df822676011ba081--libepoxy--1.5.10.ventura.bottle.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/gtkx3/manifests/3.24.41
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/808553b732670ddb6955b1843cfb0a1ee5b6324714a324739eed3b975a1bce6e--gtk+3-3.24.41.bottle_manifest.json
==> Fetching gtk+3
==> Downloading https://ghcr.io/v2/homebrew/core/gtkx3/blobs/sha256:493eb5a730ff13f34da70af0aee792ea2712b00d06fb82793dd0632ab18c3e7f
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/6c2d3c4647192e82c55e554c6e72db739560026efd77b11abc300dd14ccdd0aa--gtk+3--3.24.41.ventura.bottle.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/gobject-introspection/manifests/1.80.1
############################################################################################################################################################################### 100.0%
==> Fetching gobject-introspection
==> Downloading https://ghcr.io/v2/homebrew/core/gobject-introspection/blobs/sha256:03e5c450af9264cfe141056ccd91a5866eab01f3a15f22269092a0482a335eae
############################################################################################################################################################################### 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/py3cairo/manifests/1.26.0
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/3b75fc20df65daa3afb8b9ac23e3b95461b5876dee108ba256d36fd7a3ee6c26--py3cairo-1.26.0.bottle_manifest.json
==> Fetching py3cairo
==> Downloading https://ghcr.io/v2/homebrew/core/py3cairo/blobs/sha256:cc8d57e4c6337816342b4f3f452decf1a3199dd94419b1da1576a5b863bdda2f
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/be63d59cb861edc969609f1cad417f7fa617b63ca90935bf4191045cbc4fcafc--py3cairo--1.26.0.ventura.bottle.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/pygobject3/manifests/3.48.2
############################################################################################################################################################################### 100.0%
==> Fetching pygobject3
==> Downloading https://ghcr.io/v2/homebrew/core/pygobject3/blobs/sha256:7be31857d1eb553331fa404029bf6f510c0ea48b83395b268909a67b3eae20c7
############################################################################################################################################################################### 100.0%
==> Downloading https://ghcr.io/v2/homebrew/core/python-setuptools/manifests/69.5.1
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/f7a54d89eac25d9069fa653fbe99f04c826264b2faa00483e74c9cfb35768bcf--python-setuptools-69.5.1.bottle_manifest.json
==> Fetching python-setuptools
==> Downloading https://ghcr.io/v2/homebrew/core/python-setuptools/blobs/sha256:4bfb0b1d4e66015227843ccbaa0abed8b8b7d66667e032d0bbca1b4163a70ff9
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/ee9d0a637d1ae73c6254e32066f5222a4be778ea94c9b1b846bc055a5dde5592--python-setuptools--69.5.1.ventura.bottle.tar.gz
Warning: python-typing-extensions has been deprecated because it does not meet homebrew/core's requirements for Python library formulae!
==> Downloading https://ghcr.io/v2/homebrew/core/python-typing-extensions/manifests/4.10.0
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/b4b50328ea19f89f249ca933733e09bd49f2d358960a31bb411d568c91594817--python-typing-extensions-4.10.0.bottle_manifest.json
==> Fetching python-typing-extensions
==> Downloading https://ghcr.io/v2/homebrew/core/python-typing-extensions/blobs/sha256:a8d93b45746389920f82618145b2ba63b91b245173d7d952dda8a855f8cc68a9
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/dcf0a2dd75586da65aed7856f2f766341402b8136e5687f1daa8ecab2a6a2fa8--python-typing-extensions--4.10.0.ventura.bottle.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/six/manifests/1.16.0_4-1
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/8c932619019c6251d91f0cc7a2261022e07b740ace12ae1318f1f99e8640f0ab--six-1.16.0_4-1.bottle_manifest.json
==> Fetching six
==> Downloading https://ghcr.io/v2/homebrew/core/six/blobs/sha256:f19976d182253ff8cdf2d435b3ce6f24f2a544684baee95ab289bee753d39eb1
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/597bf69a5e2a109d38d5fb6b87fda82267c1cb208a4fd8db7d8ed287445e5458--six--1.16.0_4.all.bottle.1.tar.gz
==> Downloading https://ghcr.io/v2/homebrew/core/zlib/manifests/1.3.1
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/f68d0caf232d52f2aa586abefbbfd7e958e384d84f3967008fa83de94b5f10ae--zlib-1.3.1.bottle_manifest.json
==> Fetching zlib
==> Downloading https://ghcr.io/v2/homebrew/core/zlib/blobs/sha256:6012d7831245716d8507da3d1eb14ad274f8aa0b71b59275fe6bbbd6cebd787f
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/c6180ef9982924c8a2a6d0aea9ad26c1a12c3f780bedeaf8c1ed7b745de6d623--zlib--1.3.1.ventura.bottle.tar.gz
==> Fetching amar1729/deluge-meta/deluge-meta
==> Downloading https://github.com/Amar1729/homebrew-deluge-meta/releases/download/deluge-meta-2.1.1_4/deluge-meta-2.1.1_4.monterey.bottle.tar.gz
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/cebbe6f6e21ebf9a8af578b0e0ce3a45d70960c26fdbc060ea666ea75e2cab57--deluge-meta-2.1.1_4.monterey.bottle.tar.gz
==> Installing deluge-meta from amar1729/deluge-meta
==> Installing dependencies for amar1729/deluge-meta/deluge-meta: glib, libxrender, lzo, pixman, cairo, gdk-pixbuf, fribidi, graphite2, harfbuzz, pango, librsvg, adwaita-icon-theme, pycparser, cffi, cryptography, dbus, libxfixes, libxi, libxtst, at-spi2-core, gsettings-desktop-schemas, hicolor-icon-theme, libepoxy, gtk+3, gobject-introspection, py3cairo, pygobject3, python-setuptools, python-typing-extensions, six and zlib
==> Installing amar1729/deluge-meta/deluge-meta dependency: glib
==> Downloading https://ghcr.io/v2/homebrew/core/glib/manifests/2.80.0_2
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/bad07197fbf4d9c1af0f49bb79d89a3b369de9b90872243c8bbcfea7d4385475--glib-2.80.0_2.bottle_manifest.json
==> Pouring glib--2.80.0_2.ventura.bottle.tar.gz
🍺  /usr/local/Cellar/glib/2.80.0_2: 524 files, 34.8MB
==> Installing amar1729/deluge-meta/deluge-meta dependency: libxrender
==> Downloading https://ghcr.io/v2/homebrew/core/libxrender/manifests/0.9.11
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/134c11be8346a1b116e04983c2da6366f29c4f4c2abc17604dcdb80d0475ae9d--libxrender-0.9.11.bottle_manifest.json
==> Pouring libxrender--0.9.11.ventura.bottle.tar.gz
🍺  /usr/local/Cellar/libxrender/0.9.11: 12 files, 198.2KB
==> Installing amar1729/deluge-meta/deluge-meta dependency: lzo
==> Downloading https://ghcr.io/v2/homebrew/core/lzo/manifests/2.10
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/d4aa5b0c239912c53bc857d1012c6b7feb4acb509618f5e100f95bf8521f08e7--lzo-2.10.bottle_manifest.json
==> Pouring lzo--2.10.ventura.bottle.tar.gz
🍺  /usr/local/Cellar/lzo/2.10: 31 files, 573KB
==> Installing amar1729/deluge-meta/deluge-meta dependency: pixman
==> Downloading https://ghcr.io/v2/homebrew/core/pixman/manifests/0.42.2-1
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/2b0d8f2c63712ce503fa54309b732051a7ca5cd31bd561aa4aaba331bf26bcef--pixman-0.42.2-1.bottle_manifest.json
==> Pouring pixman--0.42.2.ventura.bottle.1.tar.gz
🍺  /usr/local/Cellar/pixman/0.42.2: 11 files, 1.3MB
==> Installing amar1729/deluge-meta/deluge-meta dependency: cairo
==> Downloading https://ghcr.io/v2/homebrew/core/cairo/manifests/1.18.0
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/1b8b7f19e38cb0e029497dfe0ad3f3557f605025c5575bdfe86c17b889c6e6d1--cairo-1.18.0.bottle_manifest.json
==> Pouring cairo--1.18.0.ventura.bottle.tar.gz
🍺  /usr/local/Cellar/cairo/1.18.0: 52 files, 2MB
==> Installing amar1729/deluge-meta/deluge-meta dependency: gdk-pixbuf
==> Downloading https://ghcr.io/v2/homebrew/core/gdk-pixbuf/manifests/2.42.11-1
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/e33b7eccf9c79f017289001438f9402b37bc235e3dc91458dd57345ad2ed1098--gdk-pixbuf-2.42.11-1.bottle_manifest.json
==> Pouring gdk-pixbuf--2.42.11.ventura.bottle.1.tar.gz
==> /usr/local/Cellar/gdk-pixbuf/2.42.11/bin/gdk-pixbuf-query-loaders --update-cache
🍺  /usr/local/Cellar/gdk-pixbuf/2.42.11: 151 files, 3.8MB
==> Installing amar1729/deluge-meta/deluge-meta dependency: fribidi
==> Downloading https://ghcr.io/v2/homebrew/core/fribidi/manifests/1.0.14
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/e6b501ebd099a81598298a9fe97eac544af056e9b6031dabf29d953431e229c1--fribidi-1.0.14.bottle_manifest.json
==> Pouring fribidi--1.0.14.ventura.bottle.tar.gz
🍺  /usr/local/Cellar/fribidi/1.0.14: 67 files, 693.9KB
==> Installing amar1729/deluge-meta/deluge-meta dependency: graphite2
==> Downloading https://ghcr.io/v2/homebrew/core/graphite2/manifests/1.3.14
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/bbb4dd2ef1846301d1eb08053e19e11ca9c780f93f4d3b2d638fd94a9bf54a0c--graphite2-1.3.14.bottle_manifest.json
==> Pouring graphite2--1.3.14.ventura.bottle.tar.gz
🍺  /usr/local/Cellar/graphite2/1.3.14: 18 files, 260.5KB
==> Installing amar1729/deluge-meta/deluge-meta dependency: harfbuzz
==> Downloading https://ghcr.io/v2/homebrew/core/harfbuzz/manifests/8.4.0
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/fd47e37d9f5b5084316ec51955470aa32e038246275985aee10b1881a4f7344b--harfbuzz-8.4.0.bottle_manifest.json
==> Pouring harfbuzz--8.4.0.ventura.bottle.tar.gz
🍺  /usr/local/Cellar/harfbuzz/8.4.0: 76 files, 9.9MB
==> Installing amar1729/deluge-meta/deluge-meta dependency: pango
==> Downloading https://ghcr.io/v2/homebrew/core/pango/manifests/1.52.2
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/214b7cb46e8e1680ea60d3578939e548b2daf2721164a11efeda086b9d494949--pango-1.52.2.bottle_manifest.json
==> Pouring pango--1.52.2.ventura.bottle.tar.gz
🍺  /usr/local/Cellar/pango/1.52.2: 68 files, 3.2MB
==> Installing amar1729/deluge-meta/deluge-meta dependency: librsvg
==> Downloading https://ghcr.io/v2/homebrew/core/librsvg/manifests/2.58.0
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/7b1ad23688052396ca6401b20573b2f69fafc28221b623268521dc99ef161ea7--librsvg-2.58.0.bottle_manifest.json
==> Pouring librsvg--2.58.0.ventura.bottle.tar.gz
==> /usr/local/opt/gdk-pixbuf/bin/gdk-pixbuf-query-loaders --update-cache
🍺  /usr/local/Cellar/librsvg/2.58.0: 22 files, 46.5MB
==> Installing amar1729/deluge-meta/deluge-meta dependency: adwaita-icon-theme
==> Downloading https://ghcr.io/v2/homebrew/core/adwaita-icon-theme/manifests/46.0
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/696462fef1577dada7d1fc747b9b32c68590e48b3d1328f7dbd07c8a90d31132--adwaita-icon-theme-46.0.bottle_manifest.json
==> Pouring adwaita-icon-theme--46.0.all.bottle.tar.gz
🍺  /usr/local/Cellar/adwaita-icon-theme/46.0: 842 files, 11.3MB
==> Installing amar1729/deluge-meta/deluge-meta dependency: pycparser
==> Downloading https://ghcr.io/v2/homebrew/core/pycparser/manifests/2.22
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/eeea4f104b2e432de9839123571bf8ac17b912aa138005099f4e97992d3e56b3--pycparser-2.22.bottle_manifest.json
==> Pouring pycparser--2.22.ventura.bottle.tar.gz
🍺  /usr/local/Cellar/pycparser/2.22: 73 files, 1.2MB
==> Installing amar1729/deluge-meta/deluge-meta dependency: cffi
==> Downloading https://ghcr.io/v2/homebrew/core/cffi/manifests/1.16.0_1-1
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/eb1e9d6f4f724404e336e6bd8af635bd56b992b0d8a67c5665a2ed7f91938b46--cffi-1.16.0_1-1.bottle_manifest.json
==> Pouring cffi--1.16.0_1.ventura.bottle.1.tar.gz
🍺  /usr/local/Cellar/cffi/1.16.0_1: 69 files, 1.1MB
==> Installing amar1729/deluge-meta/deluge-meta dependency: cryptography
==> Downloading https://ghcr.io/v2/homebrew/core/cryptography/manifests/42.0.5
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/e1105ccd58b6027d55214410024349f2af1f04f2cd983a527a6757e5fa86179f--cryptography-42.0.5.bottle_manifest.json
==> Pouring cryptography--42.0.5.ventura.bottle.tar.gz
🍺  /usr/local/Cellar/cryptography/42.0.5: 213 files, 7.9MB
==> Installing amar1729/deluge-meta/deluge-meta dependency: dbus
==> Downloading https://ghcr.io/v2/homebrew/core/dbus/manifests/1.14.10
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/2dee97539c09d15349d9935f14c299bce3e52d11fd6d399a05a5190d0e7a1028--dbus-1.14.10.bottle_manifest.json
==> Pouring dbus--1.14.10.ventura.bottle.tar.gz
==> /usr/local/Cellar/dbus/1.14.10/bin/dbus-uuidgen --ensure=/usr/local/var/lib/dbus/machine-id
🍺  /usr/local/Cellar/dbus/1.14.10: 77 files, 2.4MB
==> Installing amar1729/deluge-meta/deluge-meta dependency: libxfixes
==> Downloading https://ghcr.io/v2/homebrew/core/libxfixes/manifests/6.0.1
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/b78eeb464b4c153eb6e75a04ba240b4972cdc642e77e9c5085e186630c2c4862--libxfixes-6.0.1.bottle_manifest.json
==> Pouring libxfixes--6.0.1.ventura.bottle.tar.gz
🍺  /usr/local/Cellar/libxfixes/6.0.1: 12 files, 122KB
==> Installing amar1729/deluge-meta/deluge-meta dependency: libxi
==> Downloading https://ghcr.io/v2/homebrew/core/libxi/manifests/1.8.1
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/fd0ed36e5bf4048d89c5a4691f66ea6d88f113822452224af6ada7c579428ab0--libxi-1.8.1.bottle_manifest.json
==> Pouring libxi--1.8.1.ventura.bottle.tar.gz
🍺  /usr/local/Cellar/libxi/1.8.1: 85 files, 675.6KB
==> Installing amar1729/deluge-meta/deluge-meta dependency: libxtst
==> Downloading https://ghcr.io/v2/homebrew/core/libxtst/manifests/1.2.4
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/c3624db58ac89924c9a01073cf5dc8a8d1acbe9c3ed52af7ec5a74e92816d50c--libxtst-1.2.4.bottle_manifest.json
==> Pouring libxtst--1.2.4.ventura.bottle.tar.gz
🍺  /usr/local/Cellar/libxtst/1.2.4: 22 files, 145.5KB
==> Installing amar1729/deluge-meta/deluge-meta dependency: at-spi2-core
==> Downloading https://ghcr.io/v2/homebrew/core/at-spi2-core/manifests/2.52.0
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/32108c66e5f0d77340b05ddf14af03f14415a6c4490ade88b0e5d27edc0de1eb--at-spi2-core-2.52.0.bottle_manifest.json
==> Pouring at-spi2-core--2.52.0.ventura.bottle.tar.gz
🍺  /usr/local/Cellar/at-spi2-core/2.52.0: 199 files, 4.1MB
==> Installing amar1729/deluge-meta/deluge-meta dependency: gsettings-desktop-schemas
==> Downloading https://ghcr.io/v2/homebrew/core/gsettings-desktop-schemas/manifests/46.0
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/9b0da63c1cb9e0d2c49b1e1ab0db3278275b969537c1dc3a6d8f663115f8310e--gsettings-desktop-schemas-46.0.bottle_manifest.json
==> Pouring gsettings-desktop-schemas--46.0.ventura.bottle.tar.gz
==> /usr/local/opt/glib/bin/glib-compile-schemas /usr/local/share/glib-2.0/schemas
🍺  /usr/local/Cellar/gsettings-desktop-schemas/46.0: 103 files, 5.1MB
==> Installing amar1729/deluge-meta/deluge-meta dependency: hicolor-icon-theme
==> Downloading https://ghcr.io/v2/homebrew/core/hicolor-icon-theme/manifests/0.17
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/713c984987cfb15e35fc3fa0e951f5ab343f2154fdeab925615cb11b5bef9389--hicolor-icon-theme-0.17.bottle_manifest.json
==> Pouring hicolor-icon-theme--0.17.ventura.bottle.tar.gz
🍺  /usr/local/Cellar/hicolor-icon-theme/0.17: 7 files, 77.7KB
==> Installing amar1729/deluge-meta/deluge-meta dependency: libepoxy
==> Downloading https://ghcr.io/v2/homebrew/core/libepoxy/manifests/1.5.10
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/fd91321dc6032305294ad3c7c3851bc80f1505f18512e4753036c509ba4b4e2c--libepoxy-1.5.10.bottle_manifest.json
==> Pouring libepoxy--1.5.10.ventura.bottle.tar.gz
🍺  /usr/local/Cellar/libepoxy/1.5.10: 10 files, 2.5MB
==> Installing amar1729/deluge-meta/deluge-meta dependency: gtk+3
==> Downloading https://ghcr.io/v2/homebrew/core/gtkx3/manifests/3.24.41
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/808553b732670ddb6955b1843cfb0a1ee5b6324714a324739eed3b975a1bce6e--gtk+3-3.24.41.bottle_manifest.json
==> Pouring gtk+3--3.24.41.ventura.bottle.tar.gz
==> /usr/local/opt/glib/bin/glib-compile-schemas /usr/local/share/glib-2.0/schemas
==> /usr/local/Cellar/gtk+3/3.24.41/bin/gtk3-update-icon-cache -f -t /usr/local/share/icons/hicolor
==> /usr/local/Cellar/gtk+3/3.24.41/bin/gtk-query-immodules-3.0 > /usr/local/lib/gtk-3.0/3.0.0/immodules.cache
🍺  /usr/local/Cellar/gtk+3/3.24.41: 746 files, 55.8MB
==> Installing amar1729/deluge-meta/deluge-meta dependency: gobject-introspection
==> Downloading https://ghcr.io/v2/homebrew/core/gobject-introspection/manifests/1.80.1
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/6d5a35df34845a9b4cc3cb78447691dcaa9196bd6cac41c7b8fb880bf97e6b45--gobject-introspection-1.80.1.bottle_manifest.json
==> Pouring gobject-introspection--1.80.1.ventura.bottle.tar.gz
🍺  /usr/local/Cellar/gobject-introspection/1.80.1: 541 files, 5.8MB
==> Installing amar1729/deluge-meta/deluge-meta dependency: py3cairo
==> Downloading https://ghcr.io/v2/homebrew/core/py3cairo/manifests/1.26.0
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/3b75fc20df65daa3afb8b9ac23e3b95461b5876dee108ba256d36fd7a3ee6c26--py3cairo-1.26.0.bottle_manifest.json
==> Pouring py3cairo--1.26.0.ventura.bottle.tar.gz
🍺  /usr/local/Cellar/py3cairo/1.26.0: 37 files, 1.1MB
==> Installing amar1729/deluge-meta/deluge-meta dependency: pygobject3
==> Downloading https://ghcr.io/v2/homebrew/core/pygobject3/manifests/3.48.2
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/7df11b28a04745a1207fb9268c7a562ad7273df2e2e684198c67fb5ad237307b--pygobject3-3.48.2.bottle_manifest.json
==> Pouring pygobject3--3.48.2.ventura.bottle.tar.gz
🍺  /usr/local/Cellar/pygobject3/3.48.2: 67 files, 1.6MB
==> Installing amar1729/deluge-meta/deluge-meta dependency: python-setuptools
==> Downloading https://ghcr.io/v2/homebrew/core/python-setuptools/manifests/69.5.1
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/f7a54d89eac25d9069fa653fbe99f04c826264b2faa00483e74c9cfb35768bcf--python-setuptools-69.5.1.bottle_manifest.json
==> Pouring python-setuptools--69.5.1.ventura.bottle.tar.gz
🍺  /usr/local/Cellar/python-setuptools/69.5.1: 274 files, 3.2MB
==> Installing amar1729/deluge-meta/deluge-meta dependency: python-typing-extensions
==> Downloading https://ghcr.io/v2/homebrew/core/python-typing-extensions/manifests/4.10.0
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/b4b50328ea19f89f249ca933733e09bd49f2d358960a31bb411d568c91594817--python-typing-extensions-4.10.0.bottle_manifest.json
==> Pouring python-typing-extensions--4.10.0.ventura.bottle.tar.gz
🍺  /usr/local/Cellar/python-typing-extensions/4.10.0: 21 files, 302KB
==> Installing amar1729/deluge-meta/deluge-meta dependency: six
==> Downloading https://ghcr.io/v2/homebrew/core/six/manifests/1.16.0_4-1
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/8c932619019c6251d91f0cc7a2261022e07b740ace12ae1318f1f99e8640f0ab--six-1.16.0_4-1.bottle_manifest.json
==> Pouring six--1.16.0_4.all.bottle.1.tar.gz
🍺  /usr/local/Cellar/six/1.16.0_4: 19 files, 88.2KB
==> Installing amar1729/deluge-meta/deluge-meta dependency: zlib
==> Downloading https://ghcr.io/v2/homebrew/core/zlib/manifests/1.3.1
Already downloaded: /Users/2600box/Library/Caches/Homebrew/downloads/f68d0caf232d52f2aa586abefbbfd7e958e384d84f3967008fa83de94b5f10ae--zlib-1.3.1.bottle_manifest.json
==> Pouring zlib--1.3.1.ventura.bottle.tar.gz
🍺  /usr/local/Cellar/zlib/1.3.1: 13 files, 410.2KB
==> Installing amar1729/deluge-meta/deluge-meta
==> Pouring deluge-meta-2.1.1_4.monterey.bottle.tar.gz
==> Caveats
To successfully launch the GUI (deluge or deluge-gtk) you MUST follow the procedure
detailed in https://github.com/Amar1729/homebrew-deluge-meta/issues/23 due to an
issue with an underlying dependency.

If you do not need to GUI, no extra steps are required.

To start amar1729/deluge-meta/deluge-meta now and restart at login:
  brew services start amar1729/deluge-meta/deluge-meta
Or, if you don't want/need a background service you can just run:
  /usr/local/opt/deluge-meta/bin/deluged --do-not-daemonize --loglevel info --logfile /usr/local/var/log/deluge-meta/deluged.log
==> Summary
🍺  /usr/local/Cellar/deluge-meta/2.1.1_4: 2,546 files, 24.6MB
==> Running `brew cleanup deluge-meta`...
Disable this behaviour by setting HOMEBREW_NO_INSTALL_CLEANUP.
Hide these hints with HOMEBREW_NO_ENV_HINTS (see `man brew`).
==> Caveats
==> deluge-meta
To successfully launch the GUI (deluge or deluge-gtk) you MUST follow the procedure
detailed in https://github.com/Amar1729/homebrew-deluge-meta/issues/23 due to an
issue with an underlying dependency.

If you do not need to GUI, no extra steps are required.

To start amar1729/deluge-meta/deluge-meta now and restart at login:
  brew services start amar1729/deluge-meta/deluge-meta
Or, if you don't want/need a background service you can just run:
  /usr/local/opt/deluge-meta/bin/deluged --do-not-daemonize --loglevel info --logfile /usr/local/var/log/deluge-meta/deluged.log
```

**Comment by jmerhar** [comment]
> Hello, I have been frustrated by this problem. I have tried following all the steps outlined in the different posts of this thread without success. Whatever I do, it still goes for the latest package.
> 
> Is there any further guidance or advice that could help?
> 
> I know this will be fixed in brew eventually, but I checking everyday, so getting the workaround to _work_ would help me a lot.
> 
> Thanks!

I had the same problem. For me, the issue was that homebrew auto-updated whenever I ran the install command. I fixed it by turning auto-update off:

```
HOMEBREW_NO_INSTALL_FROM_API=1 HOMEBREW_NO_AUTO_UPDATE=1 brew install amar1729/deluge-meta/deluge-meta
```

**Comment by 2600box** [comment]
> > Hello, I have been frustrated by this problem. I have tried following all the steps outlined in the different posts of this thread without success. Whatever I do, it still goes for the latest package.
> > Is there any further guidance or advice that could help?
> > I know this will be fixed in brew eventually, but I checking everyday, so getting the workaround to _work_ would help me a lot.
> > Thanks!
> 
> I had the same problem. For me, the issue was that homebrew auto-updated whenever I ran the install command. I fixed it by turning auto-update off:
> 
> ```
> HOMEBREW_NO_INSTALL_FROM_API=1 HOMEBREW_NO_AUTO_UPDATE=1 brew install amar1729/deluge-meta/deluge-meta
> ```

Brilliant. That worked right away. Thank you!

**Comment by acuD1** [comment]
Hi, if I understand correctly, for now, we should always use `HOMEBREW_NO_INSTALL_FROM_API=1 HOMEBREW_NO_AUTO_UPDATE=1` for every brew install commands ?

Ty.

**Comment by Amar1729** [comment]
@acuD1 at least for the install commands for this formula, yes (and i think i just set `HOMEBREW_NO_AUTO_UPDATE=1` globally anyway).

**Comment by 2600box** [comment]
so, glib 2.82 is now on brew. Does this fix this issue?

https://github.com/Homebrew/homebrew-core/pull/182682

**Comment by Amar1729** [comment]
Happily, I think the updated glib does fix the issue. Waiting on linked PR to close this as it doesn't technically build right now.
